content/docs/forms.md

```diff
@@ -190,6 +190,61 @@ Overall, this makes it so that `<input type="text">`, `<textarea>`, and `<select
 ><select multiple={true} value={['B', 'C']}>
 >```
 
+## The file input Tag
+
+In HTML, a `<input type="file">` tag lets the user choose one or more files from their devices storage, allowing the app to handle them to accomplish it's goals.
+
+```html
+<input type="file">
+```
+
+In React a `<input type="file">` works similarly to a normal `<input/>` with some major differences. This input is read-only so programmatically setting the value is impossible.
+You can use the various File management API provided by Javascript to handle the files that are uploaded by the user.
+
+```javascript{5,12-18,31}
+class FileForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      file: ''
+    };
+
+    this.handleFileChange = this.handleFileChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleFileChange(e) {
+    e.preventDefault();
+
+    this.setState({
+      file: e.target.files[0]
```

Comment:
Don't think the file needs to be saved in state, unless I'm overlooking something. Probably having a `ref` to the file input and use it to access the file(s) in submit handler (or wherever required) would be a better option IMHO. See if this makes sense - https://codepen.io/tsriram/pen/qVNoZL?editors=0011

Reply:
Yeah, using the file by ref is definitly a betetr solution. My bad. Will change it ASAP

Reply:
Friendly ping 😄

Reply:
Sorry guys, been away for a while and didn't notice the progress of the issue. Seems like @vishalvrv9 has worked on a fix of my PR based on @tsriram's suggestion. Should I go ahead and fix mine or would it better to merge his?

Reply:
Yeah, had some issues so I couldn't track down my PR after submitting it but the man definitly deserves some attention for his efforts ;)